cmd/argocd/commands/app.go

```diff
@@ -2644,21 +2647,23 @@ func waitOnApplicationStatus(ctx context.Context, acdClient argocdclient.Client,
 			}
 		}
 
+		hydrationFinished := app.Status.SourceHydrator.CurrentOperation != nil && app.Status.SourceHydrator.CurrentOperation.Phase == argoappv1.HydrateOperationPhaseHydrated && app.Status.SourceHydrator.CurrentOperation.SourceHydrator.DeepEquals(app.Status.SourceHydrator.LastSuccessfulOperation.SourceHydrator) && app.Status.SourceHydrator.CurrentOperation.DrySHA == app.Status.SourceHydrator.LastSuccessfulOperation.DrySHA
```

Comment:
`app.Status.SourceHydrator.CurrentOperation` will always be nil if the source hydrator is disabled. This the CLI code, so I think it's fine to rely on the presence/absence of the status field rather than querying the API server for the feature flag state.